server/src/main/java/org/opensearch/common/metrics/ResourceTracker.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.metrics;
+
+import com.sun.management.ThreadMXBean;
+import org.opensearch.common.SuppressForbidden;
+
+import java.lang.management.ManagementFactory;
+
+/*
+    Simple tracker for CPU consumption and memory allocated by the current thread
+*/
+@SuppressForbidden(reason = "ThreadMXBean enables tracking resource consumption by a thread. "
+    + "It is platform dependent and i am not aware of an alternate mechanism to extract this info")
+public class ResourceTracker {
+
+    static ThreadMXBean threadMXBean = (ThreadMXBean) ManagementFactory.getThreadMXBean();
+    long startingAllocatedBytes;
+    long startingCPUTime;
+    long memoryAllocated;
+    long cpuTime;
+
+    public ResourceTracker() {
+        reset();
+    }
+
+    /**
+     * Takes current snapshot of resource usage by thread since the creation of this object
+     */
+    public void updateMetrics() {
+        this.memoryAllocated = threadMXBean.getThreadAllocatedBytes(Thread.currentThread().getId()) - startingAllocatedBytes;
+        this.cpuTime = threadMXBean.getCurrentThreadCpuTime() - startingCPUTime;
```

Comment:
Are these ThreadMXBean operations expensive? The alternative is to compute CPU time and memory allocated on demand, which would obviate the need for this update method and make this class a little bit simpler.

Reply:
> Are these ThreadMXBean operations expensive? 

After the improvements in https://bugs.openjdk.java.net/browse/JDK-8207266, they cost ~37ns. I had run rally benchmarks and did not notice a performance change. 

>The alternative is to compute CPU time and memory allocated on demand, which would obviate the need for this update method and make this class a little bit simpler. 

@andrross Could you please expand more on this approach. Would like to explore it.

Reply:
So the suggestion here is getting rid of the update method in `ResourceTracker` and moving the delta logic into `get..()`  - this is a good idea and i will incorporate after the integration with the [task framework(consumer)](https://github.com/opensearch-project/OpenSearch/pull/1555) as  there are no callers to `get..()` right now except the test to validate if metrics got updated. 

Post integration with task framework, this would look like - https://github.com/malpani/OpenSearch/commit/d093dc5f2a58a57a24962c97495918b4fc63ebde but i will make those in a future PR to keep the diff compact.